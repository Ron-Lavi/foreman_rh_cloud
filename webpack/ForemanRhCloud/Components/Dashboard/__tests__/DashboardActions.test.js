import { testActionSnapshotWithFixtures } from '@theforeman/test';
import API from 'foremanReact/API';
import {
  startPolling,
  stopPolling,
  fetchLogs,
  setActiveTab,
  downloadReports,
  toggleFullScreen,
} from '../DashboardActions';
import {
  pollingProcessID,
  serverMock,
  activeTab,
  accountID,
} from '../Dashboard.fixtures';

jest.mock('foremanReact/API');
API.get.mockImplementation(() => serverMock);

const runWithGetState = (state, action, params) => dispatch => {
  const getState = () => ({
    dashboard: state,
  });
  action(params)(dispatch, getState);
};

const fixtures = {
  'should startPolling': () => startPolling(accountID, pollingProcessID),
  'should fetchLogs': () =>
    runWithGetState({ activeTab: 'uploads' }, fetchLogs, accountID),
  'should stopPolling': () => stopPolling(accountID, pollingProcessID),
  'should setActiveTab': () => setActiveTab(accountID, activeTab),
  'should downloadReports': () => downloadReports(accountID),
  'should toggleFullScreen': () =>
    runWithGetState({ activeTab: 'reports' }, toggleFullScreen, accountID),
};

describe('Dashboard actions', () => {
  const { location } = window;

  beforeAll(() => {
    delete window.location;
    window.location = { href: jest.fn() };
  });

  afterAll(() => {
    window.location = location;
  });

  return testActionSnapshotWithFixtures(fixtures);
});
