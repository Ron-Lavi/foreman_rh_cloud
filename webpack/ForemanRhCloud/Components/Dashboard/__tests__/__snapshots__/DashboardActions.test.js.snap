// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard actions should downloadReports 1`] = `
Object {
  "payload": Object {
    "accountID": "some-account-ID",
  },
  "type": "INVENTORY_REPORTS_DOWNLOAD",
}
`;

exports[`Dashboard actions should fetchLogs 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "accountID": "some-account-ID",
        "activeTab": "generating",
        "logs": Array [
          "some-logs",
          "some-logs",
        ],
        "scheduled": "2019-08-21T16:14:16.520+03:00",
      },
      "type": "INVENTORY_POLLING",
    },
  ],
]
`;

exports[`Dashboard actions should setActiveTab 1`] = `
Object {
  "payload": Object {
    "accountID": "some-account-ID",
    "activeTab": "uploads",
  },
  "type": "INVENTORY_TAB_CHANGED",
}
`;

exports[`Dashboard actions should startPolling 1`] = `
Object {
  "payload": Object {
    "accountID": "some-account-ID",
    "pollingProcessID": 1,
  },
  "type": "INVENTORY_POLLING_START",
}
`;

exports[`Dashboard actions should stopPolling 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "accountID": "some-account-ID",
      },
      "type": "INVENTORY_POLLING_STOP",
    },
  ],
]
`;

exports[`Dashboard actions should toggleFullScreen 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "accountID": "some-account-ID",
        "activeTab": "generating",
      },
      "type": "INVENTORY_TOGGLE_TERMINAL_FULL_SCREEN",
    },
  ],
]
`;
